Automattic/jetpack
# Properly Gate the Paid Newsletter Feature based on Newsletter Config

Fixes https://github.com/Automattic/jetpack/issues/29693
Fixes https://github.com/Automattic/jetpack/issues/29657

## Proposed changes:

Previously we were gating the frontend components and backend code based on the following criteria:

1. Newsletter feature is enabled
2. Newsletter Plan is configured
3. Stripe KYC completed.

We need the above checks for some frontend components while other components should be displayed when the Newsletter feature is enabled. This PR resolves the issue by separating the configuration check from the feature enable check and using each check to gate components where required.

### Other information:


## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






### Before testing

1. Create a new test site on WPCOM and sandbox that test site
2. Ensure that the test site is on a  personal plan or highee
3. Ensure that you are sandboxing store by adding the following to `0-sandbox.php`:

```
define( 'USE_STORE_SANDBOX', true );
define( 'USE_BILLING_SANDBOX', true );
define( 'BILLINGDADDY_SANDBOX', rtrim( 'https://' . `hostname` ) );
```

4. Ensure that you are sandboxing  `public-api.wordpress.com`
5. Add this PR to your sandbox: `jetpack rsync jetpack [sandbox_url]:~/public_html/wp-content/mu-plugins/jetpack-plugin/production`

### Test Case A

1. Add the Subscribe block to a post
2. Verify the "Add Payments" button is displayed on the Subscribe block and in the post publish panel

### Test Case B

1. Publish a post with Paid Subscriber's access level. This can be done from the post publish panel
2. Verify the access level is saved correctly
3. Open the post in an incognito tab and verify that the post is locked
